### PR TITLE
Added subscores for pepXML to idXML conversion// MSFragger Open search features

### PR DESCRIPTION
**Pull request by tillenglert**
# Description:

Added the subscores written by MSFragger into the pepXML as Metavalues to the idXML written by the MSFraggerAdapter.

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

### CODE REVIEW

**Comment by timosachsenberg** [comment]
Ok some other test failed.
Maybe we need to check if the files used in the IDFileConverter test are really from MSFragger.
Otherwise, we might annotate MSFragger:hyperscore while it e.g. another hyperscore.

```
https://abibuilder.informatik.uni-tuebingen.de/jenkins/job/openms/job/PR/job/bldtst/compiler=appleclang-11.0.0,os_label=mojave/1171/testReport/junit/projectroot.src.tests/topp/TOPP_IDFileConverter_2_out1/

Offending lines:			(tab_width = 8, first_column = 1)

in1:  /Users/builder/jenkins/ws/openms/PR/bldtst/85c1a982/build/src/tests/topp/IDFileConverter_2_output.tmp   (line: 61, position/column: 5/34)
				<!
				 UserParam type="float" name="MSFragger:hyperscore" value="329.0"/>

in2:  /Users/builder/jenkins/ws/openms/PR/bldtst/85c1a982/source/src/tests/topp/IDFileConverter_2_output.idXML   (line: 61, position/column: 4/26)
			<!
			 /PeptideHit>
```

File: src/openms/source/FORMAT/PepXMLFile.cpp [hunk 1]
```diff
@@ -1193,6 +1193,16 @@ namespace OpenMS
         current_peptide_.setHigherScoreBetter(true);
         peptide_hit_.setMetaValue("MS:1001419", value); // def: "SpectraST spectrum score.
       }
+      else if (name == "hyperscore") // MSFragger subscore
+      {
+        value = attributeAsDouble_(attributes, "value");
+        peptide_hit_.setMetaValue("MSFragger:hyperscore", value);
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/PepXMLFile.cpp:1199]: also check (search_engine_ == "MSFragger")

Comment (timosachsenberg) [reply src/openms/source/FORMAT/PepXMLFile.cpp:1199]: because X!Tandem might also have a score with name "hyperscore"

**Comment by jpfeuffer** [comment]
What is the status here? Tests are still failing.

File: src/utils/MSFraggerAdapter.cpp [hunk 2]
```diff
@@ -791,7 +799,8 @@ const String TOPPMSFraggerAdapter::opt_out = "opt_out";
 const String TOPPMSFraggerAdapter::database = "database";
 
 // tolerance
-const String TOPPMSFraggerAdapter::precursor_mass_tolerance = "tolerance:precursor_mass_tolerance";
+const String TOPPMSFraggerAdapter::precursor_mass_tolerance_lower = "tolerance:precursor_mass_tolerance_upper";
+const String TOPPMSFraggerAdapter::precursor_mass_tolerance_upper = "tolerance:precursor_mass_tolerance_lower";
```
Comment (timosachsenberg) [inline src/utils/MSFraggerAdapter.cpp:803]: mix up upper = ... lower ?

**Comment by jpfeuffer** [comment]
src/openms/source/ANALYSIS/ID/PercolatorFeatureSetHelper.cpp:166:90: error: unused parameter ‘peptide_ids’ [-Werror=unused-parameter]

**Comment by jpfeuffer** [comment]
from GH Actions

**Comment by timosachsenberg** [comment]
tx. k fill fix

**Pull request merged**